dev_setup.sh

```diff
@@ -370,23 +370,23 @@ cd "${TOP}"
 
 # Install pep8 pre-commit hook
 HOOK_FILE="./.git/hooks/pre-commit"
-if [ ! -z ${INSTALL_PRECOMMIT_HOOK+x} ] || grep -q "MYCROFT DEV SETUP" ${HOOK_FILE}; then
-    if [ ! -f ${HOOK_FILE} ] || grep -q "MYCROFT DEV SETUP" ${HOOK_FILE} ; then
+if [[ -n ${INSTALL_PRECOMMIT_HOOK} ]] || grep -q "MYCROFT DEV SETUP" ${HOOK_FILE}; then
+    if [[ ! -f ${HOOK_FILE} ]] || grep -q "MYCROFT DEV SETUP" ${HOOK_FILE} ; then
```

Comment:
Nothing you have to fix but I wonder if it's an old bug here in the grep statement if there are spaces in the path?

Reply:
@forslund 
The `+x` replaces the contained string with an `x` IF it exists, so it does not have spaces anymore. If the check is just about empty string or not, then the contained `+x` is actually an alternative to double quoting, since it works around possible space issues. But the curly braces can be removed safely if no manipulation (like +x) is done and no literal meant characters are added to the string.

Will add a commit to make optimal bash code 🙂.

Reply:
I might have been unclear, I was reffering to the grep command part of it: `grep -q "MYCROFT DEV SETUP" ${HOOK_FILE}` will `${HOOK_FILE}` be expanded to a single argument if it the HOOK_FILE path has spaces is my question.

Reply:
@forslund 
Ah yep, theoretically doubles quotes make sense here. However the variable is just assigned the lines above, clearly without spaces.